rules/typescript/index.bzl

```diff
@@ -1,26 +1,76 @@
-load("@npm//@bazel/typescript:index.bzl", "ts_project")
-load("@npm//@bazel/esbuild:index.bzl", "esbuild")
+load("@aspect_rules_swc//swc:swc.bzl", "swc_transpiler")
 load("@build_bazel_rules_nodejs//internal/common:copy_to_bin.bzl", "copy_to_bin")
+load("@npm//@bazel/esbuild:index.bzl", "esbuild")
+load("@npm//@bazel/typescript:index.bzl", "ts_project")
+load("@npm//@bazel/jasmine:index.bzl", "jasmine_node_test")
 
-def ts_library(name, srcs, **kwargs):
-    # Copy TS sources to the bazel bin dir since TypeScript doesn't do well with
-    # source trees at multiple roots, especially in multi-bazel-workspace
-    # scenarios.
-    # See the notes here:
-    # https://github.com/aspect-build/rules_js/blob/43936ea96e085c0d6dc0f57403b0fb395d75f4f4/README.md?plain=1#L86-L100
-    # This also lets us avoid having to list out every possible bazel-out/{config}/bin
-    # dir in our tsconfig.json, as suggested here:
-    # https://bazelbuild.github.io/rules_nodejs/TypeScript.html#ts_project
-    copy_to_bin(
-        name = "%s_binsrcs" % name,
-        srcs = srcs,
-        visibility = ["//visibility:private"],
+def _swc(**kwargs):
+    swc_transpiler(
+        swcrc = "//:.swcrc",
+        **kwargs
     )
+
+def ts_library(name, srcs, **kwargs):
     ts_project(
         name = name,
         tsconfig = "//:tsconfig",
         composite = True,
-        validate = False,
-        srcs = [":%s_binsrcs" % name],
+        transpiler = _swc,
+        srcs = srcs,
+        **kwargs
+    )
+
+def ts_jasmine_node_test(name, srcs, deps = [], size = "small", **kwargs):
+    if len(srcs) != 1:
+        fail("srcs must contain exactly one TS source file")
+
+    # jasmine_node_test can't run TS tests directly, so compile the TS to JS
```

Comment:
I'd be fine to put this on hold and wait till 1.0 -- frontend development is getting pretty slow though, so in the meantime I might spend some time on splitting up some targets so that they don't invalidate so much of the build graph when changed (mainly the invocation target).